src/code/m_lights.c

```diff
@@ -0,0 +1,366 @@
+#include "global.h"
+#include "m_lights.h"
+#include "gfx.h"
+#include "macros.h"
+#include "sys_matrix.h"
+#include "overlays/gamestates/ovl_play/m_play.h"
+#include "6E0F50.h"
+
+extern Gfx D_400AA00[];
+extern Gfx D_400AA40[];
+
+static LightBuffer light_list_buf;
+
+void point_data_set(Lights* lights, s16 x, s16 y, s16 z, u8 r, u8 g, u8 b, s16 radius, s32 type) {
+    lights->type = type;
+    lights->lights.point.x = x;
+    lights->lights.point.y = y;
+    lights->lights.point.z = z;
+    Light_point_color_set(lights, r, g, b, radius);
+}
+
+void func_8009B23C_jp(Lights* lights, s16 x, s16 y, s16 z, u8 r, u8 g, u8 b, s16 radius) {
+    point_data_set(lights, x, y, z, r, g, b, radius, 0);
+}
+
+void func_8009B2A0_jp(Lights* lights, s16 x, s16 y, s16 z, u8 r, u8 g, u8 b, s16 radius) {
+    point_data_set(lights, x, y, z, r, g, b, radius, 2);
+}
+
+void Light_point_color_set(Lights* lights, u8 r, u8 g, u8 b, s16 radius) {
+    lights->lights.point.color[0] = r;
+    lights->lights.point.color[1] = g;
+    lights->lights.point.color[2] = b;
+    lights->lights.point.radius = radius;
+}
+
+void Light_diffuse_ct(Lights* lights, s8 x, s8 y, s8 z, u8 r, u8 g, u8 b) {
+    lights->type = 1;
+    lights->lights.diffuse.x = x;
+    lights->lights.diffuse.y = y;
+    lights->lights.diffuse.z = z;
+    lights->lights.diffuse.color[0] = r;
+    lights->lights.diffuse.color[1] = g;
+    lights->lights.diffuse.color[2] = b;
+}
+
+void func_8009B394_jp(LightsN* lights, u8 r, u8 g, u8 b) {
```

Comment:
This could be `LightsN_ct`

Reply:
*Could be*, do you want me to change it? I left the ones that don't match GC.